service/history/api/reapplyevents/api.go

```diff
@@ -185,9 +185,15 @@ func Invoke(
 				shard.GetLogger().Error("failed to re-apply stale events", tag.Error(err))
 				return nil, err
 			}
+			if len(reappliedEvents) == 0 {
+				return &api.UpdateWorkflowAction{
+					Noop:               true,
+					CreateWorkflowTask: false,
+				}, nil
+			}
 			return &api.UpdateWorkflowAction{
-				Noop:               true,
-				CreateWorkflowTask: false,
+				Noop:               false,
+				CreateWorkflowTask: true,
```

Comment:
I remember eventsReapplier already creates the workflow task if necessary and we need to use false here so that if all signals get reapplied are those that don't need workflow task, we don't workflow task here as well.